crates/task-impls/src/quorum_proposal.rs

```diff
@@ -739,126 +738,16 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> QuorumProposalTaskState<TYPE
-                    warn!("Leader key does not match key in proposal");
-                    return;
-                }
-
-                let justify_qc = proposal.data.justify_qc.clone();
-
-                if !justify_qc.is_valid_cert(self.quorum_membership.as_ref()) {
-                    error!("Invalid justify_qc in proposal for view {}", *view);
-                    let consensus = self.consensus.write().await;
-                    consensus.metrics.invalid_qc.update(1);
-                    return;
-                }
-
-                // Validate the upgrade certificate -- this is just a signature validation.
-                // Note that we don't do anything with the certificate directly if this passes; it eventually gets stored as part of the leaf if nothing goes wrong.
-                if let Err(e) = UpgradeCertificate::validate(
-                    &proposal.data.upgrade_certificate,
-                    &self.quorum_membership,
-                ) {
-                    warn!("{:?}", e);
-
-                    return;
-                }
-
-                // NOTE: We could update our view with a valid TC but invalid QC, but that is not what we do here
-                self.update_view(view, &event_sender).await;
-
-                let consensus = self.consensus.upgradable_read().await;
+            #[cfg(feature = "dependency-tasks")]
+            HotShotEvent::QuorumProposalValidated(proposal, _) => {
+                let new_view = proposal.view_number + 1;
 
-                // Get the parent leaf and state.
-                let parent = match consensus
-                    .saved_leaves
-                    .get(&justify_qc.get_data().leaf_commit)
-                    .cloned()
+                if let Err(e) =
+                    handle_quorum_proposal_validated(proposal, event_sender.clone(), self).await
                 {
-                    Some(leaf) => {
-                        if let (Some(state), _) =
-                            consensus.get_state_and_delta(leaf.get_view_number())
-                        {
-                            Some((leaf, Arc::clone(&state)))
-                        } else {
-                            error!("Parent state not found! Consensus internally inconsistent");
-                            return;
-                        }
-                    }
-                    None => None,
-                };
-
-                if justify_qc.get_view_number() > consensus.high_qc.view_number {
-                    if let Err(e) = self
-                        .storage
-                        .write()
-                        .await
-                        .update_high_qc(justify_qc.clone())
-                        .await
-                    {
-                        warn!("Failed to store High QC not voting. Error: {:?}", e);
-                        return;
-                    }
+                    warn!(?e, "Failed to handle QuorumProposalValidated event");
```

Comment:
This should probably be `debug`, and formatted with `{e}` rather than `?e`

Reply:
Yes! I had made this change before our discussion yesterday. Thanks for pointing this out.